COLLABORATING.md

```diff
@@ -0,0 +1,17 @@
+
+### Guide
+
+- we do trunk based development
+- `main` must never be broken or show warnings. An easy way to achieve this is to run `make tests check-size` before pushing.
```

Comment:
I'd make mention to suggest a git-hook here. We don't need to force collaborators to install one, but it might be a good idea to tell collaborators about it.
```suggestion
- `main` must never be broken or show warnings. An easy way to achieve this is to run `make tests check-size` before pushing. If you're unsure about remembering to do this, a pre-commit git hook is recommended.
 ```